v1/ast/parser_test.go

```diff
@@ -70,138 +70,1710 @@ func TestNumberTerms(t *testing.T) {
+
+				input = fmt.Sprintf(`data.%s["foo"]`, kw)
+				exp = RefTerm(VarTerm("data"), StringTerm(kw), StringTerm("foo"))
+				assertParseOneTerm(t, input, input, exp, popts)
+
+				input = "data.foo." + kw
+				exp = RefTerm(VarTerm("data"), StringTerm("foo"), StringTerm(kw))
+				assertParseOneTerm(t, input, input, exp, popts)
+
+				input = `data["foo"].` + kw
+				exp = RefTerm(VarTerm("data"), StringTerm("foo"), StringTerm(kw))
+				assertParseOneTerm(t, input, input, exp, popts)
+
+				input = kw + ".foo"
+				exp = RefTerm(VarTerm(kw), StringTerm("foo"))
+				assertParseOneTerm(t, input, input, exp, popts)
+
+				input = kw + `["foo"]`
+				exp = RefTerm(VarTerm(kw), StringTerm("foo"))
+				assertParseOneTerm(t, input, input, exp, popts)
+			})
+		}
+	})
+}
+
+func TestImportContainingKeywords(t *testing.T) {
+	for _, regoVersion := range []RegoVersion{RegoV0, RegoV1} {
+		popts := ParserOptions{RegoVersion: regoVersion}
+
+		t.Run(regoVersion.String(), func(t *testing.T) {
+			for _, kw := range KeywordsForRegoVersion(regoVersion) {
+				t.Run(kw, func(t *testing.T) {
+					// Keywords are allowed mid-path in import paths.
+
+					input := fmt.Sprintf("import data.%s.foo", kw)
+					exp := &Import{Path: RefTerm(VarTerm("data"), StringTerm(kw), StringTerm("foo"))}
+					t.Run(input, func(t *testing.T) {
+						assertParseImport(t, input, input, exp, popts)
+					})
+
+					input = fmt.Sprintf(`import data.%s["foo"]`, kw)
+					exp = &Import{Path: RefTerm(VarTerm("data"), StringTerm(kw), StringTerm("foo"))}
+					t.Run(input, func(t *testing.T) {
+						assertParseImport(t, input, input, exp, popts)
+					})
+
+					// Keywords are not allowed as the first component of import paths (only 'data, 'input', 'future', and 'rego' allowed).
+
+					input = "import " + kw
+					expErr := fmt.Sprintf(`rego_parse_error: unexpected import path, must begin with one of: {data, future, input, rego}, got: %s (hint: if this is unexpected, try updating OPA)
+	import %s
+	       ^`, kw, kw)
+					t.Run(input, func(t *testing.T) {
+						assertParseErrorContains(t, input, input, expErr, popts)
+					})
+
+					input = fmt.Sprintf("import %s.foo", kw)
+					expErr = fmt.Sprintf(`rego_parse_error: unexpected import path, must begin with one of: {data, future, input, rego}, got: %s (hint: if this is unexpected, try updating OPA)
+	import %s.foo
+	       ^`, kw, kw)
+					t.Run(input, func(t *testing.T) {
+						assertParseErrorContains(t, input, input, expErr, popts)
+					})
+
+					input = fmt.Sprintf(`import %s["foo"]`, kw)
+					expErr = fmt.Sprintf(`rego_parse_error: unexpected import path, must begin with one of: {data, future, input, rego}, got: %s (hint: if this is unexpected, try updating OPA)
+	import %s["foo"]
+	       ^`, kw, kw)
+					t.Run(input, func(t *testing.T) {
+						assertParseErrorContains(t, input, input, expErr, popts)
+					})
+
+					// Keywords are not allowed as the last component of import paths ..
```

Comment:
E.g. don't allow you to do `import data.if`, as that would create an `if` alias, which is broken unless used in a ref `if.foo`. Aliasing is fine, though, as that removes the issue: `import data.if as my_if`.

Alternatively, we could allow this, but might put users in a position that is harder to troubleshoot.